backend/src/email/utils/callback/parsers/sendgrid.ts

```diff
@@ -0,0 +1,87 @@
+import { Request } from 'express'
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+import { Ecdsa, Signature, PublicKey } from 'starkbank-ecdsa'
+import config from '@core/config'
+import logger from '@core/logger'
+import {
+  updateDeliveredStatus,
+  updateBouncedStatus,
+  updateComplaintStatus,
+} from '@email/utils/callback/update-status'
+
+const PUBLIC_KEY = PublicKey.fromPem(
+  config.get('emailCallback.sendgridPublicKey')
+)
+const SIGNATURE_HEADER = 'X-Twilio-Email-Event-Webhook-Signature'
+const TIMESTAMP_HEADER = 'X-Twilio-Email-Event-Webhook-Timestamp'
+
+type SendgridEvent = SendgridRecord[]
+
+type SendgridRecord = {
+  // the unique_args that we passed
+  message_id: string
+  // standard args
+  email: string
+  event: string
+  type: string
+  'smtp-id': string
```

Comment:
Sorry why is the property in quotes? `'smtp-id'`

Reply:
because it's not just alphanumeric? if it were `smtpId` then you wouldn't need quotes.

Reply:
Ah thanks I didn't know that